.github/workflows/test-macos-cpu.yml

```diff
@@ -0,0 +1,72 @@
+name: Unit-tests on Macos CPU
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+  workflow_dispatch:
+
+env:
+  CHANNEL: "nightly"
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        python_version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
+    uses: pytorch/test-infra/.github/workflows/macos_job.yml@main
+    with:
+      runner: macos-12
```

Comment:
I'm going to make the runner arg a matrix entry to support both mac x86 and M1 from a single workflow. (will need to see if this causes any dependency issues while running the tests)

Reply:
Will do this in a follow-up since there currently aren't M1 tests so this is slightly lower-pri.